server/server.js

```diff
@@ -1,15 +1,229 @@
+//   }
+//   const order = req.query.order;
+
+//   let orderVideos;
+//   if (order === "asc") {
+//     orderVideos = videos.sort((a, b) => a.rating - b.rating);
+//   } else {
+//     // Default to descending order if order is not specified or is invalid
+//     orderVideos = videos.sort((a, b) => b.rating - a.rating);
+//   }
+
+//   return res.json(orderVideos);
+// });
+//ordering by assending and dessending for example /videos?order=asc
+app.get("/videos", async function (req, res) {
+  const result = await db.query("SELECT * FROM videos");
+  if (result.rows.length === 0) {
+    return res.status(404).json({ error: "no videos found" });
+  }
+  const order = req.query.order;
+  let orderVideos;
+  console.log("Order:", order);
+  console.log("Before Sorting:", result.rows);
+  if (order === "asc") {
+    orderVideos = result.rows.sort((a, b) => a.rating - b.rating);
+  } else if (order === "desc") {
+    orderVideos = result.rows.sort((a, b) => b.rating - a.rating);
+  } else {
+    return res.status(400).json({ error: "Invalid order parameter" });
+  }
+  return res.status(200).json(orderVideos);
+});
+
+
+
+//adding new video
+// app.post(
+//   "/videos",
+//   [
+//     body("title", "text can't be empty").notEmpty(),
+//     body("url", "text can't be empty").notEmpty(),
+//   ],
+//   function (req, res) {
+//     const errors = validationResult(req);
+//     if (!errors.isEmpty()) {
+//       return res.status(400).send({
+//         error: errors.array(),
+//       });
+//     }
+//     const newVideo = {
+//       id: videos.length.toString(),
+//       title: req.body.title,
+//       url: req.body.url,
+//        rating:req.body.rating
+//     };
+//     videos.push(newVideo);
+//     res.status(201).json(videos);
+//   }
+// );
+
+//adding new id
+app.post(
+  "/videos",
+  [
+    body("title", "text can't be empty").notEmpty(),
+    body("url", "text can't be empty").notEmpty(),
+    body("rating", " can't be empty").notEmpty(),
+  ],
+  function (req, res) {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).send({
+        error: errors.array(),
+      });
+    }
+    const newTitle = req.body.title;
+    const newUrl = req.body.url;
+    const newRating = req.body.rating;
+
+    const query =
+      //// Include "RETURNING *" to return the newly created row
+      "INSERT INTO videos (title, url, rating) VALUES($1, $2, $3) RETURNING *";
+    db.query(query, [newTitle, newUrl, newRating], (err,result) => {
+      if (err) {
+        res.status(500).send("Internal Server Error");
+      } else {
+        const createdVideo = result.rows[0];
+        res.status(201).json(createdVideo);
+      }
+    });
+  }
+);
+
+//search video by query for example /videos/search?title=halleluja
+app.get("/videos/search", function (req, res) {
+  const searchVideo = req.query.title;
+  const filteredVideo = videos.filter((video) =>
```

Comment:
This doesn't use your database - can you rewrite this to do a SQL query (and like with the sorting, to do the filtering in SQL not JS? Again - the JS still works, but doing it in SQL will probably be faster).